### PR TITLE
New TC: Aks cloud account with mandatory and non mandatory fields

### PR DESCRIPTION
**Pull request by snigdha-px**
**What this PR does / why we need it**:
Added new testcase for azure cred 
**Which issue(s) this PR fixes** (optional)
Closes #https://purestorage.atlassian.net/browse/PB-7700

**Special notes for your reviewer**:

pass log: https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/byoc/job/Cloud_BYOC/408/console

### CODE REVIEW

File: tests/backup/backup_azure_credential_test.go [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+		azureConfigFields                                      *api.AzureConfig
+		scheduledAppContexts                                   []*scheduler.Context
+		contexts                                               []*scheduler.Context
+		appContexts                                            []*scheduler.Context
+	)
+
+	backupLocationMap1 := make(map[string]string)
+	cloudCredentialMap1 := make(map[string]string)
+	backupLocationMap2 := make(map[string]string)
+	cloudCredentialMap2 := make(map[string]string)
```
Comment (mkoppal-px) [inline tests/backup/backup_azure_credential_test.go:46]: Please move this inside `JustBeforeEach`

File: tests/backup/backup_azure_credential_test.go [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+	It("Azure cloud account with mandatory and non mandatory fields", func() {
+		ctx, err := backup.GetAdminCtxFromSecret()
+		log.FailOnError(err, "Fetching px-central-admin ctx")
+		Step("Validating applications", func() {
+			log.InfoD("Validating applications")
+			ctx, _ := backup.GetAdminCtxFromSecret()
+			controlChannel, errorGroup = ValidateApplicationsStartData(scheduledAppContexts, ctx)
+		})
+
+		Step("Creating azure cloud account and backup location with mandatory and non-mandatory fields", func() {
```
Comment (mkoppal-px) [inline tests/backup/backup_azure_credential_test.go:73]: I still think it is better to split this step. First step to create azure cloud account and backup location with mandatory fields
second step with optional fields

**Review by mkoppal-px — APPROVED** [review]
LGTM except for a couple of comments

File: tests/backup/backup_azure_credential_test.go (hunk 2 above)
Comment (snigdha-px) [reply tests/backup/backup_azure_credential_test.go]: done

File: tests/backup/backup_azure_credential_test.go (hunk 1 above)
Comment (snigdha-px) [reply tests/backup/backup_azure_credential_test.go]: done

File: tests/backup/backup_azure_credential_test.go [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+		tenantID, clientID, clientSecret, subscriptionID, accountName, accountKey = GetAzureCredsFromEnv()
+		Step("Validating applications", func() {
+			log.InfoD("Validating applications")
+			ctx, _ := backup.GetAdminCtxFromSecret()
+			controlChannel, errorGroup = ValidateApplicationsStartData(scheduledAppContexts, ctx)
+		})
+
+		Step("Creating azure cloud account and backup location with all fields", func() {
+			backupLocationProviders := GetBackupProviders()
+			for _, provider := range backupLocationProviders {
```
Comment (ak-px) [inline tests/backup/backup_azure_credential_test.go:86]: How about initializing the backupLocationProvider to Azure in the JustBeforeEach ?

File: tests/backup/backup_azure_credential_test.go [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+			restoreNameWithBkpLocationHavingMandatoryParameters = fmt.Sprintf("restore-%s-%v", backupNameWithBkpLocationHavingMandatoryParameters, RandomString(5))
+			appContextsToBackup := FilterAppContextsByNamespace(scheduledAppContexts, appNamespaces)
+			err = CreateRestoreWithValidation(ctx, restoreNameWithBkpLocationHavingMandatoryParameters, backupNameWithBkpLocationHavingMandatoryParameters, make(map[string]string), make(map[string]string), DestinationClusterName, BackupOrgID, appContextsToBackup)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Validating restore-%s", restoreNameWithBkpLocationHavingMandatoryParameters))
+		})
+
+		Step("Removing the added application cluster and registering new azure application clusters using cloud credential having only mandatory fields", func() {
+			log.InfoD("Deleting registered clusters for admin context")
+			err = DeleteCluster(SourceClusterName, BackupOrgID, ctx, true)
+			dash.VerifySafely(err, nil, fmt.Sprintf("Deleting cluster %s", SourceClusterName))
```
Comment (ak-px) [inline tests/backup/backup_azure_credential_test.go:161]: should we have wait for cluster deletion here and not fail safely .

Comment (snigdha-px) [reply tests/backup/backup_azure_credential_test.go]: corrected

File: tests/backup/backup_azure_credential_test.go (hunk 3 above)
Comment (snigdha-px) [reply tests/backup/backup_azure_credential_test.go]: Done.Since its valid only for azure ,removed the loop based on backupLocationProvider

**Review by ak-px — APPROVED** [review]
LGTM

**Pull request merged**